pvt/helpers/src/models/pools/weighted/WeightedPool.ts

```diff
@@ -47,25 +43,22 @@ import {
 } from './math';
 import { SwapKind, WeightedPoolEncoder } from '@balancer-labs/balancer-js';
 import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import BasePool from '../base/BasePool';
 
 const MAX_IN_RATIO = fp(0.3);
 const MAX_OUT_RATIO = fp(0.3);
 const MAX_INVARIANT_RATIO = fp(3);
 const MIN_INVARIANT_RATIO = fp(0.7);
 
-export default class WeightedPool {
-  instance: Contract;
-  poolId: string;
-  tokens: TokenList;
+export default class WeightedPool extends BasePool {
   weights: BigNumberish[];
   assetManagers: string[];
-  swapFeePercentage: BigNumberish;
-  vault: Vault;
   poolType: WeightedPoolType;
   swapEnabledOnStart: boolean;
   mustAllowlistLPs: boolean;
   protocolSwapFeePercentage: BigNumberish;
   managementSwapFeePercentage: BigNumberish;
+  reserveAssetManager: string;
```

Comment:
Is this meant to be here? It's not clear what this is and how it's connected to Recovery Mode.

Reply:
Yeah, I was thinking that it looked like something for the seedless LBPs. Sorry was just making sure that everything in the PR was intentional.